x/supplier/client/cli/query_claim_test.go

```diff
@@ -84,64 +95,151 @@ func createClaim(
+	supplierGenesisBuffer, err := cfg.Codec.MarshalJSON(supplierGenesisState)
+	require.NoError(t, err)
+	appGenesisState := network.ApplicationModuleGenesisStateWithAddresses(t, appAddrs)
+	appGenesisBuffer, err := cfg.Codec.MarshalJSON(appGenesisState)
+	require.NoError(t, err)
+
+	// Add supplier and application module genesis states to the network config.
+	cfg.GenesisState[types.ModuleName] = supplierGenesisBuffer
+	cfg.GenesisState[apptypes.ModuleName] = appGenesisBuffer
+
+	// Construct the network with the configuration.
 	net = network.New(t, cfg)
+	// Only the first validator's client context is populated.
+	// (see: https://pkg.go.dev/github.com/cosmos/cosmos-sdk/testutil/network#pkg-overview)
 	ctx := net.Validators[0].ClientCtx
-
-	// Prepare the keyring for the supplier account
-	kr := ctx.Keyring
-	accounts := testutil.CreateKeyringAccounts(t, kr, numClaimsPerSession)
+	// Overwrite the client context's keyring with the in-memory one that contains
+	// our pre-generated accounts.
 	ctx = ctx.WithKeyring(kr)
 
 	// Initialize all the accounts
-	for i, account := range accounts {
-		signatureSequenceNumber := i + 1
-		network.InitAccountWithSequence(t, net, account.Address, signatureSequenceNumber)
+	sequenceIndex := 1
+	for _, supplierAcct := range supplierAccts {
+		network.InitAccountWithSequence(t, net, supplierAcct.Address, sequenceIndex)
+		sequenceIndex++
+	}
+	for _, appAcct := range appAccts {
+		network.InitAccountWithSequence(t, net, appAcct.Address, sequenceIndex)
+		sequenceIndex++
 	}
 	// need to wait for the account to be initialized in the next block
 	require.NoError(t, net.WaitForNextBlock())
 
-	addresses := make([]string, len(accounts))
-	for i, account := range accounts {
-		addresses[i] = account.Address.String()
-	}
-
-	// Create one supplier
-	supplierGenesisState := network.SupplierModuleGenesisStateWithAccounts(t, addresses)
-	buf, err := cfg.Codec.MarshalJSON(supplierGenesisState)
-	require.NoError(t, err)
-	cfg.GenesisState[types.ModuleName] = buf
-
-	// Create numSessions * numClaimsPerSession claims for the supplier
+	// Create sessionCount * numClaimsPerSession claims for the supplier
 	sessionEndHeight := int64(1)
-	for sessionNum := 0; sessionNum < numSessions; sessionNum++ {
+	for sessionIdx := 0; sessionIdx < sessionCount; sessionIdx++ {
 		sessionEndHeight += numBlocksPerSession
-		sessionId := fmt.Sprintf("session_id%d", sessionNum)
-		for claimNum := 0; claimNum < numClaimsPerSession; claimNum++ {
-			supplierAddr := addresses[claimNum]
-			claim := createClaim(t, net, ctx, supplierAddr, sessionId, sessionEndHeight)
-			claims = append(claims, *claim)
-			// TODO_TECHDEBT(#196): Move this outside of the forloop so that the test iteration is faster
-			require.NoError(t, net.WaitForNextBlock())
+		// TODO_IN_THIS_COMMIT: numClaimsPerSession == supplierCount * appCount
+		for _, supplierAcct := range supplierAccts {
```

Comment:
NIT: loop through apps on the outside.

Fundamentally it results in the same thing, but logically it makes more sense given the `1 app -> N suppliers` relationship in a session.